main.tf

```
@@ -69,6 +69,12 @@ resource "aws_internet_gateway" "this" {
   tags = "${merge(map("Name", format("%s", var.name)), var.tags, var.igw_tags)}"
 }
 
+resource "aws_egress_only_internet_gateway" "this" {
+  count = "${var.enable_ipv6 && length(var.database_subnets) > 0 && length(var.private_subnets) > 0 ? 1 : 0}"
```

```suggestion
  count = "${var.enable_ipv6 && length(var.database_subnets) + length(var.private_subnets) > 0 ? 1 : 0}"
```
Otherwise this is false with `length(var.database_subnets) == 0`